src/tqec/compile/tree/tree.py

```diff
@@ -273,22 +301,46 @@ def generate_circuit(
                 plaquette. Detector computation runtime grows with this parameter,
                 so you should try to keep it to its minimum. A value too low might
                 produce invalid detectors.
-            detector_database: existing database of detectors that is used to
-                avoid computing detectors if the database already contains them.
-                Default to `None` which result in not using any kind of database
-                and unconditionally performing the detector computation.
+            detector_database: an instance to retrieve from / store in detectors
+                that are computed as part of the circuit generation. If not given,
+                the detectors are retrieved from/stored in the default location of
+                /my_detector_database.
+            database_path: specify where to save to after the calculation, when augmented.
+                As for detector_database, this defaults to /my_detector_database if
+                not specified. If detector_database is not passed in, the code attempts to
+                retrieve the database from this location.
+            do_not_use_database: if True, even the default database will not be used.
+            only_use_database: if ``True``, only detectors from the database
+                will be used. An error will be raised if a situation that is not
+                registered in the database is encountered.
             lookback: number of QEC rounds to consider to try to find detectors.
                 Including more rounds increases computation time.
 
         Returns:
             a ``stim.Circuit`` instance implementing the computation described
             by ``self``.
         """
+        # First, before we start any computations, decide which detector database to use.
+        if isinstance(database_path, str):
+            database_path = Path(database_path)
+        # If the user has passed a database in, use that, otherwise:
+        if detector_database is None:  # Nothing passed in,
+            if database_path.exists():  # look for an existing database at the path.
+                detector_database = DetectorDatabase.from_file(database_path)
+            else:  # if there is no existing database, create one.
+                detector_database = DetectorDatabase()
+        if (
+            do_not_use_database
+        ):  # override the above code and reset the database to None.
```

Comment:
Minor styling change
```suggestion
        # Override the above code and reset the database to None.
        if do_not_use_database:  
```

Reply:
I'll expand the wording to make this a bit clearer in order to move the comment to above the line.